File: src/main/java/com/xeoneux/caravansary/converter/ReservationEntityToReservationResponseConverter.java
package com.xeoneux.caravansary.converter;

import com.xeoneux.caravansary.entity.ReservationEntity;
import com.xeoneux.caravansary.response.ReservationResponse;
import org.springframework.core.convert.converter.Converter;

public class ReservationEntityToReservationResponseConverter
        implements Converter<ReservationEntity, ReservationResponse> {
    @Override
    public ReservationResponse convert(ReservationEntity source) {
        ReservationResponse reservationResponse = new ReservationResponse();
        reservationResponse.setCheckin(source.getCheckin());
        reservationResponse.setCheckout(source.getCheckout());

        return reservationResponse;
    }
}

File: src/main/kotlin/com/xeoneux/caravansary/converter/ReservationEntityToReservationResponseConverter.kt
package com.xeoneux.caravansary.converter

import com.xeoneux.caravansary.entity.ReservationEntity
import com.xeoneux.caravansary.response.ReservationResponse
import org.springframework.core.convert.converter.Converter

class ReservationEntityToReservationResponseConverter : Converter<ReservationEntity, ReservationResponse> {
    override fun convert(source: ReservationEntity): ReservationResponse {
        val reservationResponse = ReservationResponse()
        reservationResponse.checkin = source.checkin
        reservationResponse.checkout = source.checkout

        return reservationResponse
    }
}
